com.unity.render-pipelines.high-definition/Editor/AssetProcessors/MaterialPostProcessor.cs

```diff
@@ -63,9 +189,39 @@ static void RegisterUpgraderReimport()
+                        || (HDProjectSettings.pluginSubTargetLastSeenMaterialVersionsSum < HDShaderUtils.GetHDPluginSubTargetMaterialVersionsSum());
+
+                    var curHDSubTargetVersion = HDProjectSettings.hdShaderGraphLastSeenVersion;
+                    bool scanHDShaderGraphsForUpgradeNeeded = (curHDSubTargetVersion < MigrationDescription.LastVersion<ShaderGraphVersion>())
+                        || (HDProjectSettings.pluginSubTargetLastSeenSubTargetVersionsSum < HDShaderUtils.GetHDPluginSubTargetVersionsSum());
+
+                    // First test if scan for shadergraph version upgrade might be needed (triggered through their importer)
+                    // We do shadergraphs first as the HDMetaData object might need to be updated and the material update might depend
+                    // on it.
+                    // Note: this auto-update is disabled by default for now.
+                    if (s_AutoReimportProjectShaderGraphsOnVersionUpdate && scanHDShaderGraphsForUpgradeNeeded)
+                    {
+                        string commandLineOptions = System.Environment.CommandLine;
+                        bool inTestSuite = commandLineOptions.Contains("-testResults");
+                        if (!inTestSuite && fileExist && !Application.isBatchMode)
+                        {
+                            EditorUtility.DisplayDialog("HDRP ShaderGraph scan for upgrade", "The ShaderGraphs with HDRP SubTarget in your Project were created using an older version of the High Definition Render Pipeline (HDRP)." +
+                                " Unity must upgrade them to be compatible with your current version of HDRP. \n" +
+                                " Unity will re-import all of the HDRP ShaderGraphs in your project, save them to disk, and check them out in source control if needed.\n" +
+                                " Please see the Material upgrade guide in the HDRP documentation for more information.", "Ok");
+                        }
 
-                    if (curUpgradeVersion != MaterialPostprocessor.k_Migrations.Length)
+                        // When we open a project from scratch all the material have been converted and we don't need to do it two time.
+                        // However if we update with an open editor from the package manager we must call reimport.
+                        // As we can't know, we are always calling reimport resulting in unecessary work when opening a project.
+                        ReimportAllHDShaderGraphs();
+                        reimportAllHDShaderGraphsTriggered = true;
+                    }
+                    // we do else here, as we want to first call SaveAssetsToDisk() for shadergraphs, as it updates the HDProjectSettings version sentinels
+                    // (tocheck: it's probably safe to also do materials right after)
+                    else if (scanMaterialsForUpgradeNeeded)
```

Comment:
When are material upgrading happening then?

Reply:
At the next cycle of the polling